source/extensions/http/header_validators/envoy_default/http1_header_validator.cc

```diff
@@ -0,0 +1,368 @@
+      {"content-length", &Http1HeaderValidator::validateContentLengthHeader},
+  };
+
+  const auto& key_string_view = key.getStringView();
+  if (key_string_view.empty()) {
+    // reject empty header names
+    return {RejectAction::Reject, UhvResponseCodeDetail::get().EmptyHeaderName};
+  }
+
+  auto validator_it = kHeaderValidatorMap.find(key_string_view);
+  if (validator_it != kHeaderValidatorMap.end()) {
+    const auto& validator = validator_it->second;
+    return (*this.*validator)(value);
+  }
+
+  if (key_string_view.at(0) != ':') {
+    // Validate the (non-pseudo) header name
+    auto name_result = validateGenericHeaderName(key);
+    if (!name_result) {
+      return name_result;
+    }
+  } else {
+    // kHeaderValidatorMap contains every known pseudo header. If the header name starts with ":"
+    // and we don't have a validator registered in the map, then the header name is an unknown
+    // pseudo header.
+    return {RejectAction::Reject, UhvResponseCodeDetail::get().InvalidPseudoHeader};
+  }
+
+  return validateGenericHeaderValue(value);
+}
+
+::Envoy::Http::HeaderValidator::HeaderEntryValidationResult
+Http1HeaderValidator::validateResponseHeaderEntry(const HeaderString& key,
+                                                  const HeaderString& value) {
+  const auto& key_string_view = key.getStringView();
+  if (key_string_view.empty()) {
+    // reject empty header names
+    return {RejectAction::Reject, UhvResponseCodeDetail::get().EmptyHeaderName};
+  }
+
+  if (key_string_view == ":status") {
+    // Validate the :status header against the RFC valid range (100 <= status <= 599)
+    return validateStatusHeader(StatusPseudoHeaderValidationMode::ValueRange, value);
+  } else if (key_string_view == "content-length") {
+    // Validate the Content-Length header
+    return validateContentLengthHeader(value);
+  } else if (key_string_view.at(0) != ':') {
+    // Validate the generic header name.
+    auto name_result = validateGenericHeaderName(key);
+    if (!name_result) {
+      return name_result;
+    }
+  } else {
+    // The only valid pseudo header for responses is :status. If the header name starts with ":"
+    // and it's not ":status", then the header name is an unknown pseudo header.
+    return {RejectAction::Reject, UhvResponseCodeDetail::get().InvalidPseudoHeader};
+  }
+
+  // Validate the header value
+  return validateGenericHeaderValue(value);
+}
+
+::Envoy::Http::HeaderValidator::RequestHeaderMapValidationResult
+Http1HeaderValidator::validateRequestHeaderMap(RequestHeaderMap& header_map) {
+  absl::string_view path = header_map.getPathValue();
+  //
+  // Step 1: verify that required pseudo headers are present. HTTP/1.1 requests requires the
+  // :method and :path headers based on RFC 7230
+  // https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1:
+  //
+  // request-line   = method SP request-target SP HTTP-version CRLF
+  //
+  if (path.empty()) {
```

Comment:
I thought CONNECT had host only, no path?

Reply:
Yes, you're correct. I wasn't sure how http-parser handles the request-target for CONNECT requests. I just confirmed that the request-target is stored in `:authority` (not `:path`) for CONNECT requests. I'll update this logic and unit tests.